src/metric.ts

```diff
@@ -58,26 +58,37 @@ export class Metric {
     this.underlying.labels(this.labels).set(value);
   }
 
-  parse(output: any): number {
-    if (this.source.functionAbi.outputs.length === 1) {
-      const parsed = output as bigint;
-      if (parsed > Number.MAX_SAFE_INTEGER) {
-        throw new Error(`Value ${parsed} is too large to be a safe integer`);
-      }
-      return Number(parsed);
-    } else {
-      const [numerator, denominator] = output as [bigint, bigint];
-      if (denominator === BigInt(0)) {
-        return 0;
-      }
-      const precision = 1e6;
-      const value = (numerator * BigInt(precision)) / denominator;
-      if (value > Number.MAX_SAFE_INTEGER) {
+  parse(output: any, functionName: string): number {
```

Comment:
Let's actually add some tests for this function. This is a significant change.

Reply:
@bayological What would the test case be and what is a normal value for this?

Reply:
i'm all for testing but tbf this project has 0 tests atm and the function was even more opaque before this change. 
would argue for creating an issue and adding more than just this 1 test in another PR.

Reply:
yeah, just tried adding some basic test and was confronted with a typescript x eslint config issue rabbit hole that i don't think we should go down in this PR

Reply:
> @bayological What would the test case be and what is a normal value for this?

You could cover each case in the switch statement including the edge cases and error scenarios. There isn't a single 'normal value' for this.